src/pybindings/pycontroller/BUILD.gn

```diff
@@ -0,0 +1,182 @@
+  }
+
+  sources = [
+    "${root_gen_dir}/include/pybindings/pycontroller/CHIPErrorToExceptionBindings.cpp",
+    "${root_gen_dir}/include/pybindings/pycontroller/CHIPErrorToExceptionBindings.h",
+    "ControllerBindings/PyChip_Main.cpp",
+  ]
+
+  public_deps = [
+    ":build_exceptions",
+    "${chip_root}/src/app",
+    "${chip_root}/src/controller/data_model",
+    "${chip_root}/src/lib",
+    "${chip_root}/src/lib/core",
+    "${chip_root}/src/lib/mdns",
+    "${chip_root}/src/lib/support",
+    "${chip_root}/src/platform",
+    "${chip_root}/src/setup_payload",
+    "${chip_root}/src/transport",
+  ]
+  configs += [ ":controller_wno_deprecate" ]
+  ldflags = [ "-frtti" ]
+  if (current_os == "mac") {
+    ldflags += [
+      "-undefined",
+      "dynamic_lookup",
+    ]
+  }
+  if (current_os == "linux") {
+    libs = [ "python3.9" ]
+  }
+}
+pw_python_action("build_exceptions") {
+  script = "create_error_wrapper.py"
+
+  header_file = "${root_gen_dir}/include/pybindings/pycontroller/CHIPErrorToExceptionBindings.h"
+  cpp_file = "${root_gen_dir}/include/pybindings/pycontroller/CHIPErrorToExceptionBindings.cpp"
+  outputs = [
+    header_file,
+    cpp_file,
+  ]
+
+  args = [
+    "--output_cpp_file=" + rebase_path(cpp_file, root_build_dir),
+    "--output_header_file=" + rebase_path(header_file, root_build_dir),
+    "--error_header=" +
+        rebase_path("${chip_root}/src/lib/core/CHIPError.h", root_build_dir),
+    "--config_header=" +
+        rebase_path("${chip_root}/src/lib/core/CHIPConfig.h", root_build_dir),
+  ]
+}
+
+pw_python_action("pycontroller") {
+  script = "build-chip-wheel.py"
+
+  _py_manifest_files = [
+    {
+      src_dir = "."
+      sources = []
+    },
+    {
+      src_dir = target_out_dir
+      sources = [ "${target_out_dir}/pybindings/PyChip.so" ]
+    },
+    {
+      src_dir = "//"
+      sources = [ "//LICENSE" ]
+    },
+  ]
+
+  _py_manifest_file = "${target_gen_dir}/${target_name}.py_manifest.json"
+
+  inputs = []
+  _py_manifest_files_rebased = []
+  foreach(_manifest_entry, _py_manifest_files) {
+    inputs += _manifest_entry.sources
+    _py_manifest_files_rebased += [
+      {
+        src_dir = rebase_path(_manifest_entry.src_dir,
+                              get_path_info(_py_manifest_file, "dir"))
+        sources = rebase_path(_manifest_entry.sources, _manifest_entry.src_dir)
+      },
+    ]
+  }
+
+  _py_manifest = {
+    files = _py_manifest_files_rebased
+  }
+
+  write_file(_py_manifest_file, _py_manifest, "json")
+
+  _dist_dir = "${root_out_dir}/pybindings/pycontroller"
+
+  if (current_cpu == "x64") {
+    cpu_tag = "x86_64"
+  } else if (current_cpu == "arm64") {
+    cpu_tag = "aarch64"
+  } else {
+    cpu_tag = current_cpu
+  }
+
+  if (current_os == "mac") {
+    platform_tag = "macosx_" + string_replace(mac_deployment_target, ".", "_")
+  } else {
+    platform_tag = current_os
+  }
+
+  platform_tag = platform_tag + "_" + cpu_tag
+
+  tags = "cp37-abi3-" + platform_tag
```

Comment:
Unfortunately we can no longer use abi3 because pybind11 isn't compatible with it. So we'll have to compute the tags for the python headers used.